docs/docs/configuration/detectors.md

```diff
@@ -145,3 +147,38 @@ device_cgroup_rules:
 volumes:
   - /dev/bus/usb:/dev/bus/usb
 ```
+
+## Custom Detectors
+
+Custom python detector modules can be added at `/opt/frigate/frigate/detectors/plugins` in the container.
+See the below for an example implementation.
+
+```python
+from frigate.detectors.detection_api import DetectionApi
+from frigate.detectors.detector_config import BaseDetectorConfig
+from typing import Literal
+from pydantic import Extra, Field
+
+DETECTOR_KEY = "<detector_key>"
+
+# A pydantic model inheriting from `BaseDetectorConfig`
+# Must implement the `type` attribute as below
+# Can add any number of fields to be passed from 
+# configuration to the constructor of your detector.
+class CustomDetectorConfig(BaseDetectorConfig):
+    type: Literal[DETECTOR_KEY]
+    custom_field: str = Field(default="value", title="Custom field description")
+
+
+# The custom detector class must inherit from `DetectionApi`
+# and must implement the `type_key`, `__init__`, 
+# and `detect_raw` methods as below
+class CustomDetector(DetectionApi):
+    type_key = DETECTOR_KEY
+
+    def __init__(self, detector_config: CustomDetectorConfig):
+      ...
+
+    def detect_raw(self, tensor_input):
+      return <list of detections>
+```
```

Comment:
This doesn't belong here. These are the user docs, this is something that may belong in the contribution section

Reply:
My thought process on it is that it could enable users to provide their own implementations by just bind mounting a .py file in, if they so desired. It's an advanced scenario sure, but not implausible?

Reply:
This isn't something that I think we would want to support officially though. If someone wants to add a new detector in a PR happy to help but, we don't want a user expecting support in doing so. 

And a user adding this advanced of a feature is a "developer" anyway

Reply:
Like I said I think it's fine I'm the contributions section of the docs, just not the user part 👍